workspace/src/app/views/shared/SearchBar/SearchBar.tsx

```diff
@@ -29,45 +30,24 @@ export function SearchBar({
     focusOnCreation = false,
     ...otherProps
 }: IProps) {
-    const [searchInput, setSearchInput] = useState(textQuery);
     const [t] = useTranslation();
 
     const emptySearchMessage = otherProps.emptySearchInputMessage
         ? otherProps.emptySearchInputMessage
         : t("form.field.searchField", "Enter search term");
-    useEffect(() => {
-        setSearchInput(textQuery);
-    }, [textQuery]);
 
-    const handleSearchChange = (e) => {
-        // when input is empty then apply filter
-        if (e.target.value === "" && searchInput) {
-            setSearchInput("");
-            onSearch("");
-        } else {
-            setSearchInput(e.target.value);
-        }
-    };
-
-    const onClearanceHandler = () => {
-        setSearchInput("");
-        onSearch("");
-    };
-
-    const handleSearchEnter = () => {
-        onSearch(searchInput);
-    };
+    const { query, onChange, onEnter, onClear } = useSearch(onSearch);
```

Comment:
The delay should be configurable, <= 0 for disable.

Reply:
i'm guessing you mean >= 0, in that case, it's already added to the hook, where the default is 500 ms